app-android/src/main/java/com/ustadmobile/port/android/view/CourseDiscussionDetail2Fragment.kt

```diff
@@ -0,0 +1,355 @@
+            .width(120.dp),
+        elevation = null,
+        colors = ButtonDefaults.buttonColors(
+            backgroundColor = colorResource(R.color.secondaryColor),
+            contentColor = Color.Transparent,
+            disabledBackgroundColor = Color.Transparent,),
+    ) {
+        Row (
+            horizontalArrangement = Arrangement.End) {
+            Image(
+                painter = painterResource(id = R.drawable.ic_add_black_24dp),
+                contentDescription = null,
+                modifier = Modifier.width(25.dp),
+                colorFilter = ColorFilter.tint(color = Color.Black))
+            Spacer(modifier = Modifier.width(5.dp))
+            Text(
+                text = stringResource(R.string.post),
+                color = Color.Black,
+                textAlign = TextAlign.Center,
+                style = Typography.h5
+            )
+        }
+    }
+}
+
+@Composable
+private fun PostDetailItem(post: DiscussionPostWithDetails){
+    Row(
+        modifier = Modifier
+            .padding(8.dp)
+            .fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically
+    ){
+        val context = LocalContext.current
+        val datePosted = remember { DateFormat.getDateFormat(context)
+            .format(Date(post.discussionPostStartDate ?: 0)).toString() }
+
+        val fullName= post.authorPersonFirstNames + " " + post.authorPersonLastName
+        UstadMessageField(
+            imageId = R.drawable.ic_person_black_24dp,
+            mainText = fullName,
+            secondText = post.discussionPostTitle ?: "",
+            thirdText = post.postLatestMessage ?: "",
+            fourthText = datePosted,
+            fifthText = post.postRepliesCount.toString() + " replies",
+            thirdTextIcon = R.drawable.ic_baseline_sms_24
+        )
+    }
+}
+
+
+@Composable
+private fun CourseDiscussionDetailScreen(
+    uiState: CourseDiscussionDetailUiState = CourseDiscussionDetailUiState(),
+    onClickAddPost: () -> Unit = {},
+    onClickPost: (DiscussionPostWithDetails) -> Unit = {},
+    onClickDeletePost: (DiscussionPostWithDetails) -> Unit = {},
+){
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState())
+    ) {
+        Text(
+            uiState.courseDiscussion?.courseDiscussionTitle?:"",
+            style = Typography.body1,
+            modifier = Modifier.padding(8.dp)
+        )
+        Spacer(modifier = Modifier.height(10.dp))
+        Text(
+            uiState.courseDiscussion?.courseDiscussionDesc?:"",
+            style = Typography.body1,
+            modifier = Modifier.padding(8.dp)
+        )
+        Spacer(modifier = Modifier.height(10.dp))
+        Text(stringResource(R.string.posts),
+            style = Typography.h4,
+            modifier = Modifier.padding(8.dp))
+
+
+        Box(modifier = Modifier.weight(1f)){
+            PostsDetail(uiState.posts, onClickPost)
```

Comment:
I think it would be better to use the Jetpack compose ListItem instead of creating something additional here. ListItem will automatically apply the correct padding, spacing, typography, click ripple effect etc for a material design list. It has been used in other screens. I also notice that right now the onClickPost is not really connected anywhere, so clicking won't do anything. Something like this:

```
ListItem(
   modifier = modifier.clickable {
      onClickPost(post)
   },
   icon = {
      //Avatar icon here
   },
   text = { 
      Text(postNames)
   },
   secondaryText = {
      Column {
         Text(discussionPostTitle)
         Text(postLatestMessage)
      }
   },
   singleLineSecondaryText = false,
   trailing = {
      Column {
          Text(date)
          Text(stringResource(R.string.num_replies, post.numReplies))
       }
   }
)
```